### PR TITLE
Improved background customization options for CTABanner

### PR DESCRIPTION
**Pull request by rezrah**
## Summary

Resolves https://github.com/primer/brand/issues/756
Resolves https://github.com/primer/brand/issues/752

Adds background customization parity between `BreakoutBanner` and `CTABanner`

🔗 [Preview](https://primer-36159649ce-26139705.drafts.github.io/brand/storybook/?path=/story/components-ctabanner-features--custom-background-colors)

## List of notable changes:

- Added new props
  - `backgroundColor`
  - `backgroundImageSrc`
  - `backgroundImageSize`
  - `backgroundImagePosition`  
- Added tests (visual + unit)
- Added relevant Storybook examples
- Updated documentation usage examples

## What should reviewers focus on?

- Do documentation examples look okay?
- Are we happy with the image examples used?

## Steps to test:

1. [Storybook](https://primer-36159649ce-26139705.drafts.github.io/brand/storybook/?path=/story/components-ctabanner-features--custom-background-colors)
2. [Documentation](https://primer-36159649ce-26139705.drafts.github.io/brand/components/CTABanner/react)

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] UI Changes contain new visual snapshots (generated by adding `update snapshots` label to the PR)
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:
Visual-Comparison-CTABanner-CTABanner-Responsive-Background-Image-Dark-1-linux

### CODE REVIEW

File: apps/docs/scripts/components-with-animation.js
Comment (danielguillan) [inline apps/docs/scripts/components-with-animation.js:1]: Not sure why this keeps happening…

File: packages/react/src/CTABanner/CTABanner.tsx [hunk 1]
```diff
@@ -13,58 +13,130 @@ import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/compone
...
+     * Optional, custom background image
+     */
+    backgroundImageSrc?: string | ResponsiveBackgroundImageSrcMap
+    /**
+     * Optional, custom background position
+     */
+    backgroundImagePosition?: string | ResponsiveBackgroundImagePositionMap
+    /**
+     * Optional, custom background size
+     */
```
Comment (danielguillan) [inline packages/react/src/CTABanner/CTABanner.tsx:62]:
```suggestion
    /**
     * Optional, custom background color.
     */
    backgroundColor?: BackgroundColors | ResponsiveBackgroundColorMap
    /**
     * Optional, custom background image.
     */
    backgroundImageSrc?: string | ResponsiveBackgroundImageSrcMap
    /**
     * Optional, custom background position.
     */
    backgroundImagePosition?: string | ResponsiveBackgroundImagePositionMap
    /**
     * Optional, custom background size.
     */
```

Nitpick, for consistency.

File: packages/react/src/CTABanner/CTABanner.tsx [hunk 2]
```diff
@@ -13,58 +13,130 @@ import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/compone
...
+    backgroundImageSize?: string | ResponsiveBackgroundImageSizeMap
+    /**
+     * Escape-hatch for inserting custom React components.
+     * Warning:
+     *   This prop isn't advertised in our docs but remains part of the public API for edge-cases.
+     *   Need to use this prop? Please check in with #primer-brand first to confirm correct usage.
+     */
```
Comment (danielguillan) [inline packages/react/src/CTABanner/CTABanner.tsx:69]: ❤️

File: apps/docs/scripts/components-with-animation.js
Comment (rezrah) [reply apps/docs/scripts/components-with-animation.js:1]: I'll look into it 👍

**Pull request merged**